src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -77,52 +78,127 @@ public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimi
                 bool success = _txFilterPipeline.Execute(tx, parent);
                 if (!success) continue;
 
-                if (tx.SupportsBlobs)
+                foreach (Transaction blobTx in PickBlobTxsBetterThanCurrentTx(selectedBlobTxs, tx, comparer))
                 {
-                    if (blobGasPrice.IsZero)
-                    {
-                        ulong? excessBlobGas = BlobGasCalculator.CalculateExcessBlobGas(parent, _specProvider.GetSpec(parent));
-                        if (excessBlobGas is null)
-                        {
-                            if (_logger.IsTrace) _logger.Trace($"Declining {tx.ToShortString()}, the specification is not configured to handle shard blob transactions.");
-                            continue;
-                        }
-                        if (!BlobGasCalculator.TryCalculateBlobGasPricePerUnit(excessBlobGas.Value, out blobGasPrice))
-                        {
-                            if (_logger.IsTrace) _logger.Trace($"Declining {tx.ToShortString()}, failed to calculate blob gas price.");
-                            continue;
-                        }
-                    }
+                    yield return blobTx;
+                }
 
-                    int txAmountOfBlobs = tx.BlobVersionedHashes?.Length ?? 0;
+                if (_logger.IsTrace) _logger.Trace($"Selected {tx.ToShortString()} to be potentially included in block.");
 
-                    if (blobGasPrice > tx.MaxFeePerBlobGas)
-                    {
-                        if (_logger.IsTrace) _logger.Trace($"Declining {tx.ToShortString()}, blob gas fee is too low.");
-                        continue;
-                    }
+                selectedTransactions++;
+                yield return tx;
+            }
 
-                    if (BlobGasCalculator.CalculateBlobGas(blobsCounter + txAmountOfBlobs) >
-                        Eip4844Constants.MaxBlobGasPerBlock)
-                    {
-                        if (_logger.IsTrace) _logger.Trace($"Declining {tx.ToShortString()}, no more blob space.");
-                        continue;
-                    }
+            if (selectedBlobTxs.Count > 0)
+            {
+                foreach (Transaction blobTx in selectedBlobTxs)
+                {
+                    yield return blobTx;
```

Comment:
not incrementing `checkedTransactions` and `selectedTransactions` for blob transactions?

Reply:
There are separate `checkedBlobTransactions` and `selectedBlobTransactions`